cmd/thanos/query_frontend.go

```diff
@@ -76,6 +76,16 @@ func registerQueryFrontend(app *extkingpin.App) {
 	cmd.Flag("query-range.split-interval", "Split query range requests by an interval and execute in parallel, it should be greater than 0 when query-range.response-cache-config is configured.").
 		Default("24h").DurationVar(&cfg.QueryRangeConfig.SplitQueriesByInterval)
 
+	cmd.Flag("query-range.split-threshold", "Split query range requests which duration are over this threshold. Using this parameter is not allowed with query-range.split-interval. "+
+		"One should also set query-range.split-min-horizontal-shards to a value greater than 1 to enable splitting.").
+		Default("0").DurationVar(&cfg.QueryRangeConfig.QuerySplitThresholdInterval)
+
+	cmd.Flag("query-range.max-split-interval", "Split query range requests using this interval. If the query duration is shorter than this value, then use query-range.split-min-horizontal-shards to split query.").
+		Default("0").DurationVar(&cfg.QueryRangeConfig.MaxQuerySplitInterval)
+
+	cmd.Flag("query-range.split-min-horizontal-shards", "Split queries in at least this amount of vertical shards, only when query duration is below query-range.max-split-interval.").
```

Comment:
I think `query-range.horizontal-shards` is more appropriate because this is not the minimum, but rather the exact number of shards in which we split [[1]](https://github.com/thanos-io/thanos/pull/5658/files#diff-fc3f06fdf01b64920948c18215315d4578a715da88fa1b75c75e73f922cb9d88R251).